source/magpie/group.d

```diff
@@ -463,6 +463,146 @@ public:
+            pos[1] = p;
+        }
+
+        static if(is(T[0] == void))
+        {
+            assert(elements.data.length == elementCountTill[pos[0] + 1] - elementCountTill[pos[0]],
+                "Size of data doesn't match size of group column");
+            static foreach(i; 0.. GrpRowType.length)
+            {
+                if(i == pos[1])
+                {
+                    foreach(j; elementCountTill[pos[0]] .. elementCountTill[pos[0] + 1])
+                    {
+                        import std.variant: VariantException;
+                        try
+                        {
+                            data[i][j] = elements.data[j - elementCountTill[pos[0]]].get!(GrpRowType[i]);
+                        }
+                        catch(VariantException e)
+                        {
+                            data[i][j] = cast(GrpRowType[i])elements.data[j - elementCountTill[pos[0]]].get!(double);
+                        }
+                    }
+                }
+            }
+        }
+        else
+        {
+            assert(elements.data.length == GrpRowType.length, "Size of data doesn't match size of group column");
+            static foreach(i; 0.. GrpRowType.length)
+            {
+                static foreach(i; 0 .. GrpRowType.length)
+                    data[i][elementCountTill[pos[0]] + pos[1]] = elements.data[i];
+            }
+        }
+    }
+
+    /// Short Hand Binary operation
+    void opIndexOpAssign(string op, T...)(Axis!T elements, string[] groupTitle, string[] index, int axis = 1)
+    {
+       /*
+        * This code looks very much like the code above
+        * Initially this could be written as ofIndexAssign(opIndex[...indexes] + element, ...indexes)
+        * The problem here is that opIndex will create a new Axis struct. This additional allocation
+        * can be avoided by rewriting the implementation above to account for everything.
+        */
+
+        int[2] pos;
+        int p = getGroupPosition(groupTitle);
+        assert(p > -1, "Group index out of bound");
+        pos[0] = p;
+
+        if(axis)
+        {
+            p = positionInGroup!(1)(index, pos[0]);
+            assert(p > -1, "Index out of bound");
+            pos[1] = p;
+        }
+        else
+        {
+            p = positionInGroup!(0)(index, pos[0]);
+            assert(p > -1, "Index out of bound");
+            pos[1] = p;
+        }
+
+        static if(is(T[0] == void))
+        {
+            assert(elements.data.length == elementCountTill[pos[0] + 1] - elementCountTill[pos[0]],
+                "Size of data doesn't match size of group column");
+            static foreach(i; 0.. GrpRowType.length)
+            {
+                if(i == pos[1])
+                {
+                    foreach(j; elementCountTill[pos[0]] .. elementCountTill[pos[0] + 1])
+                    {
+                        import std.variant: VariantException;
+                        try
```

Comment:
you should check the variants [`.type`](https://dlang.org/phobos/std_variant.html#.VariantN.type) and see if it matches `typeid(GrpRowType[i])`, rather than use a `try ... catch`

Reply:
:+1: 
Addressed in 6517b4a